Content type updates for Framework migration guide (#602)
hashicorp/web-unified-docs: content/terraform-plugin-framework/v1.15.x/docs/plugin/framework/migrating/attributes-blocks/force-new.mdx

# Migrating attribute ForceNew triggers

This page explains how to migrate ForceNew triggers from SDKv2 to the framework.

## Background

> In Terraform, sometimes a resource must be replaced when the value of an
> attribute changes. In SDKv2, this is accomplished via the `ForceNew` field. In
> the framework, implement the same behavior via a `RequiresReplace` plan
> modifier. Refer to [Plan Modification - Attribute Plan
> Modification](/terraform/plugin/framework/resources/plan-modification#attribute-plan-modification)
> in the framework documentation for more details about plan modifiers.

Reviewer: ```suggestion
In Terraform, sometimes a resource must be replaced when the value of an
attribute changes. To do this in SDKv2, you use the `ForceNew` field. To do this 
in the framework, implement the same behavior with a `RequiresReplace` plan
modifier. Refer to [Plan Modification - Attribute Plan
Modification](/terraform/plugin/framework/resources/plan-modification#attribute-plan-modification)
in the framework documentation for more details about plan modifiers.
```

## Migrating ForceNew triggers

> In SDKv2, setting the `ForceNew` field on an attribute's `schema.Schema`
> triggers a replace (i.e., a destroy-create cycle) whenever the attribute's value
> is changed.

Reviewer: ```suggestion
In SDKv2, when you set the `ForceNew` field on an attribute's `schema.Schema`
it triggers a replace, such as a destroy-create cycle, whenever the attribute's value
is changed.
```
